src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -149,11 +158,24 @@ export class ElasticsearchService implements CoreService<InternalElasticsearchSe
       },
     };
 
+    const esNodesCompatibility$ = pollEsNodesVersion({
+      callWithInternalUser: adminClient.callAsInternalUser,
+      log: this.log,
+      ignoreVersionMismatch: config.ignoreVersionMismatch,
+      esVersionCheckInterval: config.healthCheckDelay.asMilliseconds(),
+      kibanaVersion: this.kibanaVersion,
+    }).pipe(publishReplay(1));
+
+    this.subscriptions.esNodesCompatibility = (esNodesCompatibility$ as ConnectableObservable<
+      unknown
+    >).connect();
```

Comment:
NIT: is there no way to avoid the force-cast by returning 'correct' type from `pollEsNodesVersion` ?

Reply:
The type problem comes from `pipe(publishReplay(1))` the `publishReplay` should change the return type to a `ConnectableObservable` but doesn't.